src/views/MapPage.vue

```diff
@@ -66,20 +90,50 @@ export default {
         subTitle,
       };
     },
+    reservedFilter() {
+      return this.$route.params.editmode === 'edit' ? 1 : 0;
+    },
+    reservedStreets() {
+      return this.$store.getters.GET_RESERVED_STREETS;
+    },
   },
   methods: {
-    pushStreet(street) {
-      if (this.streets.includes(street)) {
+    pushStreet(street, selection) {
+      console.log(selection);
+      if (this.streetsToReserve.includes(street)
+      || this.streetsToUnreserve.includes(street)
+      || this.streetsToComplete.includes(street)) {
         this.$bvModal.show('street-in-list-modal');
-      } else {
-        this.streets.push(street);
+        return;
+      }
+      if (selection === 'reserve') {
+        this.streetsToReserve.push(street);
+      } else if (selection === 'unreserve') {
+        this.streetsToUnreserve.push(street);
+      } else if (selection === 'complete') {
+        this.streetsToComplete.push(street);
       }
     },
+    getFids() {
+      const fids = [];
+      this.reservedStreets.forEach((street) => {
+        fids.push(street.FID);
+      });
+      return fids;
+    },
     async reserveStreets() {
       // reserve streets backend call
       // .then(
       this.$bvModal.show('street-confirmation-modal');
```

Comment:
Is there anything that's preventing these three methods from being connected?

Reply:
Not as it stands. Once this has been connected to the backend then there will be three different ajax calls so I figured 3 separate methods might be better than one large method with a switch statement

Reply:
Nice, I agree

Reply:
I meant like implementing the requests to the backend itself